homeassistant/components/websocket_api/automation.py

```diff
@@ -138,8 +150,51 @@ def _get_automation_component_domains(
     return domains
 
 
+def _get_automation_component_lookup_table(
+    hass: HomeAssistant,
+    component_type: str,
+    component_descriptions: Mapping[str, Mapping[str, Any] | None],
+) -> _AutomationComponentLookupTable:
+    """Get a dict of automation components keyed by domain, along with the total number of components.
+
+    Returns a cached object if available.
+    """
+
+    try:
+        cache = hass.data[AUTOMATION_COMPONENT_LOOKUP_CACHE]
+    except KeyError:
+        cache = hass.data[AUTOMATION_COMPONENT_LOOKUP_CACHE] = {}
+
+    if (cached := cache.get(component_type)) is not None:
+        cached_descriptions, cached_lookup = cached
+        if cached_descriptions is component_descriptions:
+            _LOGGER.debug(
```

Comment:
Maybe add a log when we don't use the cache instead and assume we use the cache if there's no log? If speed is important we should avoid logging.